### PR TITLE
Keycloak task does not update config

### PR DESCRIPTION
**Pull request by j-zimnowoda**
Although we could see that task discovers existing configuration it has never update an existing configuration in some cases.
Read more: #41 

Instead of relaying on  getting HTTP 409 response for POST method and creating noise in keycloak log, I decided to check if resource exists and based on the result to either update or create the resource.

### CODE REVIEW

**Review by Dunky13 — APPROVED** [review]

**Review by Morriz — CHANGES_REQUESTED** [review]
I might be nitpicking here, but I just like to be thorough

File: src/tasks/keycloak/keycloak.ts [hunk 1]
```diff
@@ -73,106 +76,116 @@ async function main(): Promise<void> {
...
   const client = realmConfig.createClient()
-  const clientExists = await doApiCall(errors, 'Creating otomi client', () =>
-    clients.realmClientsPost(env.KEYCLOAK_REALM, client),
-  )
-  if (clientExists) {
+  const allClients = (await doApiCall(errors, 'Getting otomi client', () =>
+    clients.realmClientsGet(env.KEYCLOAK_REALM),
+  )) as Array<ClientRepresentation>
+
+  if (allClients.some((el) => el.name === client.name)) {
```
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply src/tasks/keycloak/keycloak.ts:169]: Find also implies that first one matching is good enough .

After reading https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find I do not see a difference except `some` returns boolean and `find` returns integer.

File: src/tasks/keycloak/keycloak.ts [hunk 2]
```diff
@@ -73,106 +76,116 @@ async function main(): Promise<void> {
...
-        )
-      }
-    }),
-  )
 
+  const existingMappers = (await doApiCall(errors, `Getting role mappers`, () =>
+    providers.realmIdentityProviderInstancesAliasMappersGet(env.KEYCLOAK_REALM, env.IDP_ALIAS),
+  )) as IdentityProviderMapperRepresentation[]
+
+  idpMappers.map(async (idpMapper) => {
```
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply src/tasks/keycloak/keycloak.ts]: done

File: src/tasks/keycloak/keycloak.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply src/tasks/keycloak/keycloak.ts:169]: not do

**Review by Morriz — APPROVED** [review]

**Pull request merged**